torch/testing/_internal/common_distributed.py

```diff
@@ -877,6 +888,66 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for DDP based Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def backend(self, device) -> str:
+        if "cuda" in device:
+            return "nccl"
+        elif "hpu" in device :   # intel gaudi
+            return "hccl"
+        else :
+            return "gloo"
+
+    def device_handle(self, device) :
```

Comment:
Can you add comment and typing?

Reply:
added in commit: 65e16deb07c5ca9acef323058a99d26e89633acb